lib/elements/steel_reinforcement/base.rb

```diff
@@ -0,0 +1,14 @@
+module Elements
+  class Base
+    def initialize(number:, material:)
+      @diameter = diameter.to_f
+      @number = number
+      @material = material
+      @start_hook = nil
+      @end_hook = nil
+
+      rebar_data = DB::Base::STANDARD_REBAR.find { |bar_data| bar_data['number'] == number }
+      @diameter = rebar_data['diameter'].to_f
+    end
```

Comment:
1. Given this approach, you are setting the `start_hook` and `end_hook` attributes for both `Rebar` and `RebarHook`.  It looks like these attributes should be specific for `Rebar` only.
2. Why are you setting `@diameter = diameter.to_f` first and then `@diameter = rebar_data['diameter'].to_f`?
3. I think we should isolate the dependency to `DB::Base::STANDARD_REBAR`, maybe we can have a class method here that is called something like `find_standard_rebar(number:)`, which does this lookup of line 10. Something like this:
```ruby
def self.find_standard_rebar(number:)
  DB::Base::STANDARD_REBAR.find { |rebar_data|  rebar_data['number'] == number }
end
```

Reply:
what do you think:

at Elements::SteelReinforcement::Base constructor, 
```ruby
module Elements
  module SteelReinforcement
    class Base
      def initialize(number:, material:, standard_bars:)
        @number = number
        @material = material
        @start_hook = nil
        @end_hook = nil

        rebar_data = standard_bars.find_standard_rebar(number:)
        @diameter = rebar_data['diameter'].to_f
      end
    end
  end
end
```

at DB::Base
```ruby
module DB
  module Base
    STANDARD_REBAR = YAML.load_file('lib/db/rebars.yml')

    def self.find_standard_rebar(number:)
      STANDARD_REBAR.find { |rebar_data| rebar_data['number'] == number }
    end
  end
end
```

so, instantiating an Elements::SteelReinforcement::Rebar 
```ruby
Elements::SteelReinforcement::Rebar.new(number:, material: steel, standard_bars: DB::Base)
```

Reply:
Sounds good, I just have 2 small sueggestions:
1. Let's call it `standard_rebars` instead of `standard_bars`. Bar !== Rebar
2. Set the `standard_bars` argument as optional, leaving `DB::Base` as default. That way we don't need to inject it every time we instantiate a rebar.